### PR TITLE
fix: use correct runfiles path for venv import paths

### PR DESCRIPTION
**Pull request by wade-arista**
In `_make_import_path`, `ctx.workspace_name` is always `_main` when evaluating the deps of targets in the main repo. This includes deps that are in external repos, e.g.

    importer = use_extension("//:import.bzl", "importer")
    use_repo(importer, "myrepo")

with

    py_test(..., deps = ["@myrepo//foo"])

In this case, we actually want to use the label's `workspace_name` attr which will be the same as the runfiles path that we need for the venv import paths.

Also added a new e2e test that creates this external repo and verifies the fix. Without the fix in place, the test does fail as described in the e2e/repository-rule-deps/README.md "Repro" section.

---

### Test plan
- New test case added in `e2e/repository-rule-deps`.
- Manually tested against bazel 6.5 and 7.0.2.

Fixes: #276

### CODE REVIEW

**Comment by wade-arista** [comment]
I would like to see that the new e2e/repository-rule-deps is properly executed via the new "repository-rule-deps" check before this merges as well. I don't expect that I have permissions to run this, though.

**Review by alexeagle — APPROVED** [review]
Wow, nicely documented!

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -30,6 +30,13 @@ jobs:
           {"folder": ".", "bzlmodEnabled": true}
         ]
 
+  repository-rule-deps:
```
Comment (alexeagle) [inline .github/workflows/ci.yaml:33]: this doesn't need a new job, you can just add to the `folders` array of the `test` job above

**Comment by scasagrande** [comment]
thanks @wade-arista , I was in the process of fixing this same issue, but you did a much better job than I had time for 😄

**Comment by wade-arista** [comment]
> thanks @wade-arista , I was in the process of fixing this same issue, but you did a much better job than I had time for 😄

sure thing! Sorry, I had meant to post a [link to discourse](https://bazelbuild.slack.com/archives/CA306CEV6/p1709085258803439) where we were discussing a PR, but then I forgot.

Hopefully this is the same thing fix you had in mind?

**Comment by scasagrande** [comment]
Yeah I believe this covers my case but I will have to double check just to make sure. I had a quick fix for my testing so I'll just double check that it covers my case.

I did notice a further issue in my testing. After patching importing a py_libary from another workspace, I saw that the dependencies of this library not being picked up correctly. I'll take a look if that is still the case and raise a separate issue if so.

**Pull request merged**